src/modules/deal-folder/deal-folder.service.ts

```diff
@@ -0,0 +1,175 @@
+  }
+
+  private constructRequestBodyToUpdateFileInfo(documentType: string): {
+    Title: string;
+    Document_x0020_Status: string;
+  } {
+    const { documentTitle, documentTypeId } = DOCUMENT_TYPE_MAPPING[documentType];
+
+    return {
+      Title: documentTitle,
+      Document_x0020_Status: DOCUMENT_X0020_STATUS,
+      [this.config.estoreDocumentTypeIdFieldName]: documentTypeId,
+    };
+  }
+
+  private async constructUrlToUpdateFileInfo(
+    fileName: string,
+    dealId: string,
+    buyerName: string,
+    ukefSiteId: string,
+  ): Promise<{ urlToUpdateFileInfo: string; fileLeafRef: string }> {
+    const listId = await this.getResourceIdByName(ukefSiteId, CASE_LIBRARY.LIST_NAME, ENUMS.SHAREPOINT_RESOURCE_TYPES.LIST);
+
+    const webUrlRegExpForFile = this.constructWebUrlRegExpForFile(fileName, dealId, buyerName, ukefSiteId);
+    const { itemId, fileLeafRef } = await this.getItemIdAndFileLeafRefByWebUrlRegExp(ukefSiteId, listId, webUrlRegExpForFile);
+
+    const urlToUpdateFileInfo = `sites/${this.config.ukefSharepointName}:/sites/${ukefSiteId}:/lists/${listId}/items/${itemId}/fields`;
+
+    return {
+      urlToUpdateFileInfo,
+      fileLeafRef,
+    };
+  }
+
+  private constructWebUrlRegExpForFile(fileName: string, dealId: string, buyerName: string, ukefSiteId: string): RegExp {
+    const encodedBuyerName = encodeURIComponent(buyerName);
+    const encodedDealId = encodeURIComponent(dealId);
+    const encodedFileDestinationPath = `${encodedBuyerName}/D${encodedDealId}`;
+    const encodedFileName = encodeURIComponent(fileName);
+
+    return new RegExp(
+      `https://${this.config.ukefSharepointName}/sites/${ukefSiteId}/${CASE_LIBRARY.LIST_NAME}/${encodedFileDestinationPath}/.*${encodedFileName}`,
+    );
+    // Using a reg exp is necessary because Sharepoint temporarily prepends a code (which appears to always start with '~tmp') to the file name in the web url.
+  }
+
+  private getItemIdAndFileLeafRefByWebUrlRegExp(ukefSiteId: string, listId: string, webUrlRegExp: RegExp): Promise<{ itemId: string; fileLeafRef: string }> {
+    return this.graphService
+      .get<{ value: { webUrl: string; id: string; fields: { FileLeafRef: string } }[] }>({
+        path: `sites/${this.config.ukefSharepointName}:/sites/${ukefSiteId}:/lists/${listId}/items`,
+        expand: 'fields',
+      })
+      .then((response) => response.value)
+      .then((list) => list.find((item) => webUrlRegExp.test(item.webUrl))) // This is necessary because the 'filter' query parameter does not currently support the 'webUrl' field on lists/list items (see comment from Microsoft employee https://learn.microsoft.com/en-us/answers/questions/980144/graph-api-sharepoint-list-filter-by-createddatetim).
```

Comment:
Is this comment still 100% accurate? Isn't it also because we need to use regex, or does `$filter` support regex too?

Reply:
I've now removed the regex as it is no longer needed after the 'await' fix yesterday, so this comment is 100% accurate.